### ISSUE TITLE
Typescript Context for Apollo Server

### ISSUE THREAD

**Issue by tnolan8** [jkettmann/graphql-passport#34]
Hey, I'm trying to make the switch over to using Passport to verify email/password logins using `GraphQLLocalStrategy` but I'm running into an issue with defining the _type_ of my context. Previously my context looked like:

```javascript
export interface Context {
    req: Request;
    res: Response;
    prisma: PrismaClient;
    smtpTransport: any;
};
```
And I'm now trying to update the context like: 
```javascript
export interface MyContext extends PassportContext<Context, ExpressRequest>{}
```
And build my context as follows: 
```javascript
const server = new ApolloServer({
        schema,
        playground: {
            settings: {
                "request.credentials": "same-origin",
            },
        },
        context: ({ req, res }): MyContext =>
            buildContext({
                req,
                res,
                prisma,
                smtpTransport,
            }),
    });
```
But I keep running into `Property 'logOut' is missing in type 'Context<Request<ParamsDictionary>>' but required in type 'MyContext'`.

How can I update the context correctly? Thank you!

**Comment by jkettmann** [comment]
@gforge can you help out on this?

**Comment by gforge** [comment]
Sure, I'll try to have a look at this.

**Comment by jkettmann** [comment]
Thanks @gforge :smile:

**Comment by gforge** [comment]
Sorry for the delayed response. The project where I had been using this package was put on hold and I got some new test-errors after updating the packages. First, if anyone stumbles upon issues with your Request not being valid, make sure that you have the correct definitions (see [this issue](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/40138)) by deleting your *lock*-files and re-installing everything. 

I'm not sure that I actually understand the original code example. The definition for the `PassportContext` requires 4 arguments:

```ts
export declare type PassportContext<
  UserObjectType extends {}, 
  Credentials extends {}, 
  AuthInfoTemplate extends {} = {}, 
  Request extends object = ExpressRequest
> = SharedPassportContext<UserObjectType, Credentials, AuthInfoTemplate, Request>;
```

In the example provided there are only two arguments provided, `Context` and `ExpressRequest`. My definitions look something like this:


```ts
export interface Credentials {
  username: string;
  password: string;
}

declare interface MyContext extends PassportContext<UserModelInterface, Credentials, {}, express.Request> {
  dataSources: MyDataSourcesInterface;
}
```

As of [TS 2.2](https://github.com/Microsoft/TypeScript/pull/13604) there you can add the *prisma* and *smtpTransport* just as I did with `dataSources` even though the original context is a type and not an interface. If you want it stricter then perhaps an [intersection](https://www.typescriptlang.org/docs/handbook/advanced-types.html#intersection-types) is a better choice.

**Comment by sebastianplsim** [comment]
@gforge Thanks 💯

**Comment by dgurns** [comment]
Yep https://github.com/DefinitelyTyped/DefinitelyTyped/issues/40138#issuecomment-612402184 fixed it for me
